### PR TITLE
fix: add patch indicator to prevent crashes

### PR DESCRIPTION
**Pull request by Tormak9970**

### CODE REVIEW

File: src/utils/react.ts [hunk 1]
```diff
@@ -46,13 +46,17 @@ export function fakeRenderComponent(fun: Function, customHooks: any = {}): any {
 }
 
 export function wrapReactType(node: any, prop: any = 'type') {
-  return (node[prop] = { ...node[prop] });
+  return node[prop].__DECKY_WRAPPED ? node[prop] : (node[prop] = { ...node[prop], __DECKY_WRAPPED: true });
```
Comment (AAGaming00) [inline src/utils/react.ts:49]:
```suggestion
  return node[prop]?.__DECKY_WRAPPED ? node[prop] : (node[prop] = { ...node[prop], __DECKY_WRAPPED: true });
```

File: src/utils/react.ts [hunk 2]
```diff
@@ -46,13 +46,17 @@ export function fakeRenderComponent(fun: Function, customHooks: any = {}): any {
...
 
 export function wrapReactClass(node: any, prop: any = 'type') {
-  const cls = node[prop];
-  const wrappedCls = class extends cls {};
-  return (node[prop] = wrappedCls);
+  if (node[prop].__DECKY_WRAPPED) {
+    return node[prop];
+  } else {
+    const cls = node[prop];
+    const wrappedCls = class extends cls { __DECKY_WRAPPED = true; };
```
Comment (AAGaming00) [inline src/utils/react.ts:57]: shouldn't __DECKY_WRAPPED be static to make this work (or is it already?)

Comment (Tormak9970) [reply src/utils/react.ts:57]: shoot, yea it should. will change in a sec

File: src/utils/react.ts (hunk 1 above)
Comment (Tormak9970) [reply src/utils/react.ts]: done. also switched to if statement since the class wrap uses one

**Review by AAGaming00 — APPROVED** [review]

**Pull request merged**